test/test_svd.cc

```diff
@@ -208,7 +216,61 @@ void test_svd_work( Params& params, bool run )
         }
     }
 
-    if (check || ref) {
+    if (check && (wantu || wantvt)) {
+        // Residual matrix.
+        slate::Matrix<scalar_t> R( min_mn, min_mn, nb, p, q, MPI_COMM_WORLD );
+        R.insertLocalTiles();
+
+        if (wantu) {
+            //==================================================
+            // Test results by checking orthogonality of U
+            //
+            //      || I - U^H U ||_1
+            //     ------------------- < tol * epsilon
+            //              N
+            //==================================================
+            slate::set( zero, one, R ); // identity
+            auto UH = conj_transpose( U );
+            slate::gemm( -one, UH, U, one, R );
+            params.ortho_U() = slate::norm( slate::Norm::One, R ) / n;
+            params.okay() = params.okay() && (params.ortho_U() <= tol);
+        }
+
+        if (wantvt) {
+            //==================================================
+            // Test results by checking orthogonality of VT
+            //
+            //      || I - V V^H ||_1
+            //     ------------------- < tol * epsilon
+            //              N
+            //==================================================
+            slate::set( zero, one, R ); // identity
+            auto V = conj_transpose( VT );
+            slate::gemm( -one, VT, V, one, R );
+            params.ortho_V() = slate::norm( slate::Norm::One, R ) / n;
+            params.okay() = params.okay() && (params.ortho_V() <= tol);
```

Comment:
Should okay be initialized to `true` somewhere above?  I think the default is -1.

Reply:
Probably that would be more robust. Usually the first one (as in line 370) is
```
    params.okay() = (error < tol);
```
and subsequent ones are
```
    params.okay() = params.okay() && (error < tol);
```
Setting it = true and consistently using the 2nd syntax would avoid issues with reordering or if some check is skipped.

Reply:
Set to true at top.